### ISSUE TITLE
'api_version' get default value (v1) unless last alias is used

### ISSUE THREAD

**Issue by jobcespedes** [ansible-collections/cloud.common#100]
##### SUMMARY

When turbo is enable, a module parameter with aliases is not recognized and its default value is used instead. However, its last alias is. For example: k8s_info paramater `api_version`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

module

##### ANSIBLE VERSION

```paste below
ansible 2.9.23
  config file = None
  configured module search path = ['/var/home/job/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /var/home/job/.local/lib/python3.9/site-packages/ansible
  executable location = /var/home/job/.local/bin/ansible
  python version = 3.9.7 (default, Aug 30 2021, 00:00:00) [GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

##### COLLECTION VERSION

Cloud.common v2.1.0

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

Fedora 34

##### STEPS TO REPRODUCE

1. Run example playbook against a k8s: `ansible-playbook play.yaml`

```yaml
- name: test alias
  hosts: localhost
  gather_facts: false
  collections:
    - kubernetes.core
  environment:
    ENABLE_TURBO_MODE: 1
  tasks:
    - name: create obj with version different than v1 (default)
      k8s:
        state: present
        definition:
          apiVersion: batch/v1beta1
          kind: CronJob
          metadata:
            name: deleteme
            namespace: default
          spec:
            schedule: "*/1 * * * *"
            jobTemplate:
              spec:
                template:
                  spec:
                    containers:
                    - name: hello
                      image: busybox
                      imagePullPolicy: IfNotPresent
                      command:
                      - /bin/sh
                      - -c
                      - date; echo Hello from the Kubernetes cluster
                    restartPolicy: OnFailure

    - name: THIS FAIL (No alias), get obj
      retries: 1
      delay: 1
      ignore_errors: true
      until:
        - (wait_failed_task.resources | first).status is defined
      k8s_info:
        api_version: batch/v1beta1
        kind: CronJob
        namespace: default
        name: deleteme
      register: wait_failed_task

    - name: get obj
      retries: 1
      delay: 1
      until:
        - (wait_ok_task.resources | first).status is defined
      k8s_info:
        # api_version: batch/v1beta1
        api: batch/v1beta1 # alias of api_version
        kind: CronJob
        namespace: default
        name: deleteme
      register: wait_ok_task

    - name: THIS FAIL (not the last alias?), delete obj
      ignore_errors: true
      k8s:
        # api_version: batch/v1beta1
        version: batch/v1beta1 # alias of api_version
        kind: CronJob
        namespace: default
        name: deleteme
        state: absent

    - name: delete obj
      k8s:
        # api_version: batch/v1beta1
        api: batch/v1beta1 # alias of api_version
        kind: CronJob
        namespace: default
        name: deleteme
        state: absent
```

##### EXPECTED RESULTS

Api version is given the value set by `api_version`

##### ACTUAL RESULTS

Api version gets default value ("v1") even though we are setting it to a different one. 

```paste below
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [test alias] **********************************************************************************************************************

TASK [create obj with version different than v1 (default)] *****************************************************************************
changed: [localhost]

TASK [THIS FAIL (No alias), get obj] ***************************************************************************************************
FAILED - RETRYING: THIS FAIL (No alias), get obj (1 retries left).
fatal: [localhost]: FAILED! => {"api_found": false, "attempts": 1, "changed": false, "msg": "Failed to find API for resource with apiVer
sion \"v1\" and kind \"CronJob\"", "resources": []}
...ignoring

TASK [get obj] *************************************************************************************************************************
ok: [localhost]

TASK [THIS FAIL (not the last alias?), delete obj] *************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to find exact match for v1.CronJob by [kind, name, singularName, shortN
ames]"}
...ignoring

TASK [delete obj] **********************************************************************************************************************
changed: [localhost]

PLAY RECAP *****************************************************************************************************************************
localhost                  : ok=5    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=2   
```

**Comment by alinabuzachis** [comment]
FYI @gravesm. I guess this issue should be migrated under https://github.com/ansible-collections/kubernetes.core.

**Comment by jobcespedes** [comment]
This may be a duplicate of #91. A fix has been added after v2.1.0

**Comment by jobcespedes** [comment]
Any idea when a new version including #91 fix will be released?
